### ISSUE TITLE
[Bug]: Misleading error on invalid topic name when using rust client.

### ISSUE THREAD

**Issue by gibbz00** [infinyon/fluvio#3521]
**What happened**

When running CLI: 

```sh
fluvio topic create dectest0f4d91a3a9104cf18c1c989a115812e9-some-very-long-topic-name

"Invalid argument: Invalid Topic name dectest0f4d91a3a9104cf18c1c989a115812e9-some-very-long-topic-name. Name exceeds max characters allowed 63."
```

When using rust the client: 
```
`Result::unwrap()` on an `Err` value: FluvioAnyhow(Invalid topic name: topic name may only include lowercase letters (a-z), numbers (0-9), and hyphens (-).)
```

**Expected behavior**

Error message should point out exceeded character limit.

**Possible starting point**

https://github.com/infinyon/fluvio/blob/990892b09b60e11ed4d6cb4fb82557de8980ba1a/crates/fluvio-sc-schema/src/lib.rs#L61-L63

**Environment:**
 - OS: Linux, Mac
 - Fluvio Crate Version: 0.20
 - Fluvio Cluster Version: 
 ```
Release Channel      : stable
Fluvio CLI           : 0.10.14
Fluvio CLI SHA256    : 09e9c102bfe90639696b3b7f102a7bae4c965d3077d46abc23049a4dceec45cf
Fluvio channel frontend SHA256 : f0baffcd126e1114679a19b1ca27345a2376a82adf911451be6c83ecc52d9c28
fluvio_git_hash="18505ad85609f2d1c1f9d6a203178127d97af073"
Fluvio Platform      : 0.10.14 (k3d-unigis-ci)
Git Commit           : 18505ad85609f2d1c1f9d6a203178127d97af073
=== Plugin Versions ===
Fluvio Runner (fluvio-run)     : 0.0.0
Infinyon Cloud CLI (fluvio-cloud) : 0.2.11
```

**Comment by jsat97** [comment]
@digikata 
Could you please look at these changes to see if they're ok ?
https://github.com/jsat97/fluvio/tree/invalid-topic-name-error-rust-client
I was trying to test it following [this ](https://github.com/infinyon/fluvio/blob/master/DEVELOPER.md#building-and-running-fluvio-cluster-from-source-code-for-local-binaries) doc.
1. ran "make" in fluvio root dir of my checked out repo.
2. set aliases
3. I see the following when I start the cluster ( flvd cluster start --local --develop)
`✅ Local Cluster initialized
🖥️  Trying to connect to SC: localhost:9003 121 seconds elapsed /                                                  2023-11-02T18:52:15.692823Z ERROR fluvio_cluster::start::common: fail to connect to sc at: localhost:9003
Timed out when waiting for SC service
`
4, helm list
`
jai@ubuntu:~/fluvio$ helm list
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART             APP VERSION
fluvio-sys      default         1               2023-11-02 18:42:43.871846359 +0000 UTC deployed        fluvio-sys-0.9.13 0.11.0-dev-1
`
5. Where do I find the rust client as per the bug? The repo -- https://github.com/infinyon/flv-client-rust
is archived. Do we have another ?

**Comment by digikata** [comment]
re: 4. The rust client in this case is the `fluvio` cli binary, so if you do a `fluvio topic create dectest0f4d91a3a9104cf18c1c989a115812e9-some-very-long-topic-name` with your built cli binary, you should be able to cause the error as well as test a fix. (as long as you get a fluvio cluster working).

**Comment by jsat97** [comment]
You have 2 errors in your reported output. The first one shows "Invalid Topic name" as well as the error on the limit of characters.
the 2nd one shows the error on invalid characters.
I thought these were produced differently (are cli and rust client the same ?)
The error from fluvio appears to come from -- [here ](https://github.com/infinyon/fluvio/blob/4d4151ac80f439a45d0c0ea0af1675ab875490db/crates/fluvio-cli/src/client/topic/create.rs#L171)
The error is printed the same as before with my new binary (no changes to above code)
`
jai@ubuntu:~/fluvio$  flvd  topic create dectest0f4d91a3a9104cf18c1c989a115812e9-some-very-long-topic-name
Invalid argument: Invalid Topic name dectest0f4d91a3a9104cf18c1c989a115812e9-some-very-long-topic-name. Name exceeds max characters allowed 63
`

**Comment by digikata** [comment]
re: 3.  Depending on the kubernetes you are using locally, sometimes it's a little different to access the cluster.  e.g. minikube, the node port is exposed directly, but not not all k8 installations or configs work with node port.  So more diagnosis on the kubernetes setup might be needed.

At the master line right now, we have an alpha/beta  non-kubernetes linked 'local' cluster config, so it might worth trying that out too as the topic bug of the issue might be reproducible with that configuration too. (support docs have not be updated for that yet though).

**Comment by jsat97** [comment]
I dont have the cluster start problem in (3). Like I said last, I was able to bring up the cluster  - but need clarification on the two error cases shown in the report -
1. When running CLI:
2. When using rust the client:
Are these 2 different cases ? In (1), the error is reported correctly.  so nothing to fix if this is the fluvio client.
(2) -> how to repro this.

**Comment by digikata** [comment]
So technically there is a fluvio client crate which is something modular you can use to build into various other software - e.g. web servers or other backends, language bindings, etc.  And there is the fluvio cli, which makes extensive use of that same fluvio-client crate. 

If the error is reproduced using the cli, there is further diagnosis in this issue to figure out where the source of the error is - could be in the cli handling, but without digging further into it myself, I would guess it's deeper, in the client or possibly even in the cluster logic.

**Comment by sehz** [comment]
I think this is due fact CLI is remapping error.  Ideally, error from SC should be sought of truth

**Comment by jsat97** [comment]
I just saw your comments (they didn't show up until I refreshed the page!)
> If the error is reproduced differently in one context vs other,
understand. The fluvio command (if this is called the "client") is giving the correct output (error does say > 63 chars)
The 2nd case (cli ? ) is the one I am unable to reproduce.
@sehz 
> I think this is due fact CLI is remapping error.
The mapping is on the line I've shown above (create.rs)
There is apparently another  case where `validate_resource_name` is called and the error returned is mapped incorrectly.  I may have already fixed this in my [PR ](https://github.com/jsat97/fluvio/tree/invalid-topic-name-error-rust-client)
but don't know how to test it.
